src/muiThemeable.js

```diff
@@ -1,15 +1,14 @@
 import React from 'react';
-import DefaultRawTheme from './styles/raw-themes/light-raw-theme';
-import ThemeManager from './styles/theme-manager';
+import getMuiTheme from './styles/getMuiTheme';
 
 function getDisplayName(WrappedComponent) {
   return WrappedComponent.displayName || WrappedComponent.name || 'Component';
 }
 
 export default function muiThemeable(WrappedComponent) {
-  const MuiComponent = (props, {muiTheme = ThemeManager.getMuiTheme(DefaultRawTheme)}) => {
-    return <WrappedComponent {...props} muiTheme={muiTheme} />;
-  };
+  function MuiComponent(props, {muiTheme = getMuiTheme()}) {
```

Comment:
In our case it probably doesn't matter too much if its a function or class, it'd be a matter of style. I think for react-redux it was necessary because they needed access to component lifecycle methods. If we needed that as well, I'd say we should use a class but using function vs class shouldn't have any practical implications so I'm open to either (though personally I tend to write things as functions until I need them to be classes).

> Could you explain the benefit of this approach? I can't see it :grin:.

No problem! :smile: My main intentions behind that other PR were the following:

1. Removes the need for `ThemeManager`. It's one less thing user needs to know how to use. I think it'd be more ideal if they knew they could just define their `baseTheme` as a raw JSON object and pass it directly to a ThemeProvider component, and our library knows how to properly merge and/or setup the baseTheme. 
2. I think it makes more sense that the code that maps the `baseTheme` configuration for a component's section in the `muiTheme` belongs with the corresponding `Component.jsx` file, not separately in `./styles/getMuiTheme.js`
3. I think it _might_ make more sense that a component decorated with `muiTheme` properties that are relevant for it, not the whole muiTheme object. And in addition, these properties could be documented and possible validated against used `PropTypes`

I feel like I'm not presenting my arguments very clearly :stuck_out_tongue_closed_eyes: if you'd like me to elaborate, I can't do so on the #2509 PR.